test-framework/junit5/src/main/java/io/quarkus/test/junit/DisabledOnIntegrationTest.java

```diff
@@ -36,4 +36,18 @@
      * Reason for disabling this test
      */
     String value() default "";
+
+    /**
+     * The types of Quarkus application produced by the build for which this test is disabled.
+     * For example, if the tests are supposed to be disabled for built containers only,
+     * {@code ArtifactType.CONTAINER} would be set.
+     */
+    ArtifactType[] forArtifactTypes() default { ArtifactType.ALL };
+
+    enum ArtifactType {
+        ALL,
+        JAR,
```

Comment:
Mind adding javadocs to the corresponding type-strings (like `fast-jar`, `uber-jar`, etc)?

Reply:
I don't really want to do this, as these strings are internal and not meant to be used by anyone but the testing facility

Reply:
Meant the values for [`quarkus.package.type`](https://quarkus.io/guides/maven-tooling#fast-jar)

Reply:
That information isn't available at this point and in any case, for tests it should not make a difference - I mean in what scenario would want to only not execute a test if it's for an `uber-jar` for example?

Reply:
Thought it could be useful information for users to associate `quarkus.package.type=fast-jar` with the `JAR` enum value.

Reply:
(javadoc)